test/services/sorting/post_sorting_service_test.dart

```diff
@@ -0,0 +1,178 @@
+      final referencePosition = referencePost.location.geoPoint;
+      final position = post.location.geoPoint;
+
+      return Geolocator.distanceBetween(
+        referencePosition.latitude,
+        referencePosition.longitude,
+        position.latitude,
+        position.longitude,
+      );
+    }
+
+    void testOption(
+      PostSortOption option,
+      String expectedBehaviourStr,
+      double Function(PostFirestore, PostFirestore) expectedBehaviour, {
+      bool forcePositiveScore = false,
+    }) {
+      test(
+        "${option.name} score $expectedBehaviourStr",
+        () {
+          final referencePost = postGenerator.generatePostAt(userPosition0);
+          final referenceScore =
+              option.scoreFunction(referencePost, userPosition0);
+
+          final actualScoreDifference = posts
+              .map(
+                (post) =>
+                    referenceScore - option.scoreFunction(post, userPosition0),
+              )
+              .map(
+                (difference) =>
+                    forcePositiveScore ? difference.abs() : difference,
+              );
+
+          final expectedScoreDifference = posts.map(
+            (post) => expectedBehaviour(referencePost, post),
+          );
+          expect(actualScoreDifference, expectedScoreDifference);
+        },
+      );
+    }
+
+    testOption(
+      PostSortOption.hottest,
+      "increases by 1 every upvote, decreases by 1 every day",
+      (referencePost, post) =>
+          upvoteDifference(referencePost, post) -
+          dayDifference(referencePost, post),
+    );
+
+    testOption(
+      PostSortOption.latest,
+      "increases by 1 every every day",
+      dayDifference,
+    );
+
+    testOption(
+      PostSortOption.nearest,
+      "decreases by 1 every meter",
+      positionDifference,
+      forcePositiveScore: true,
+    );
+
+    testOption(
+      PostSortOption.top,
+      "increases by 1 every upvote",
+      upvoteDifference,
+    );
+  });
+
+  group("Empty onTop attribute", () {
+    for (final option in PostSortOption.values) {
+      test("Correct sort on option ${option.name}", () {
+        final sorted = sortingService.sort(
+          posts,
+          option,
+          userPosition0,
+        );
+
+        expect(sorted, unorderedEquals(posts));
+
+        final scores = sorted.map(
+          (post) => option.scoreFunction(post, userPosition0),
+        );
+        expectSorted(scores, option.sortIncreasing);
+      });
+    }
+  });
+
+  group("Nonempty onTop attribute", () {
+    final positionsToPutOnTop =
+        GeoPointGenerator.generatePositions(userPosition0, 5, 0);
+
+    /// The posts that we will ask to put on top, if in the list
+    final postsToPutOnTop =
+        postGenerator.generatePostsAtDifferentLocations(positionsToPutOnTop);
+
+    /// The posts that will actually be in the list to be sorted
+    final existingPostsToPutOnTop = postsToPutOnTop.take(3).toList();
```

Comment:
For similar reasons to my previous comment, I think we could make those `late` and initialize them in the `setUp`.
```suggestion
      late List<GeoPoint> positionsToPutOnTop;
      late List<PostFirestore> postsToPutOnTop;
      late List<PostFirestore> existingPostsToPutOnTop;

      setUp(() {
        positionsToPutOnTop =
            GeoPointGenerator.generatePositions(userPosition0, 5, 0);

        postsToPutOnTop = postGenerator
            .generatePostsAtDifferentLocations(positionsToPutOnTop);

        existingPostsToPutOnTop = postsToPutOnTop.take(3).toList();
      });
```

Reply:
Fixed in d6c9c4ecd56bf2b95cdcdb3926ee37184c6624b8.